crates/bdk/src/wallet/mod.rs

```diff
@@ -1685,52 +1758,62 @@ impl<D> Wallet<D> {
     /// transactions related to your wallet into it.
     ///
     /// [`commit`]: Self::commit
-    pub fn apply_update(&mut self, update: Update) -> Result<(), UpdateError>
+    pub fn apply_update(&mut self, update: Update) -> Result<bool, UpdateNotConnectedError>
     where
-        D: persist::PersistBackend<KeychainKind, ConfirmationTime>,
+        D: PersistBackend<ChangeSet>,
     {
-        let changeset = self.keychain_tracker.apply_update(update)?;
-        self.persist.stage(changeset);
-        Ok(())
+        let mut changeset: ChangeSet = self.chain.apply_update(update.chain)?.into();
+        let (_, derivation_additions) = self
+            .indexed_graph
+            .index
+            .reveal_to_target_multi(&update.keychain);
+        changeset.append(derivation_additions.into());
+        changeset.append(self.indexed_graph.apply_update(update.graph).into());
+
+        let changed = !changeset.is_empty();
+        if changed {
+            self.persist.stage(changeset);
```

Comment:
No need for this branch. If it is empty it'll just be a noop.

Reply:
Removed all unnecessary branches.